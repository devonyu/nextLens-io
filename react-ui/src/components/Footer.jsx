import React, { Component } from 'react';
import { Button, Container, Grid, Segment, Icon, Divider } from 'semantic-ui-react'

export default class Footer extends Component {
    render() {
        return (
            <Container id='footer' fluid >
              <Grid columns={3} relaxed>
                <Grid.Column id='ft1' inverted="true">
                    <Segment basic inverted={true}>
                    <Icon name='camera retro'></Icon><a href='https://www.nextlens.io'>NextLens.io</a>
                    </Segment>
                </Grid.Column>

                <Divider vertical fitted/>

                <Grid.Column id='ft2' inverted="true">
                    <Segment basic inverted={true}>Terms && Usage</Segment>
                </Grid.Column>

                <Divider vertical fitted/>
                
                <Grid.Column id='ft3' inverted="true">
                    <Segment basic inverted={true}>
                        <Button circular color='facebook' icon='facebook' href='https://www.facebook.com/nextlens.io/'/>
                        <Button circular color='grey' icon='github' href='https://github.com/devonyu/nextLens-io'/>
                        <Button circular color='twitter' icon='twitter' href='https://twitter.com/nextlens'/>
                        <Button circular color='instagram' icon='instagram' href='https://www.instagram.com/nextlens.io/'/>
                    </Segment>
                </Grid.Column>
                
            </Grid>
            </Container>    
        )
    }
}
