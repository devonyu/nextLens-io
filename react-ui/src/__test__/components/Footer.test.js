import React from 'react';
import Footer from '../../components/Footer';
import renderer from 'react-test-renderer';

test('Can see footer', () => {
  const component = renderer.create(<Footer />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
