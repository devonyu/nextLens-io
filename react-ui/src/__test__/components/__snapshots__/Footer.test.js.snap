// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Can see footer 1`] = `
<div
  className="sc-bdVaJa gzfAUc"
>
  <div
    className="sc-bwzfXH cCmMUB"
  >
    <a
      className="sc-htpNat gEKCED"
      href="https://nextlens.io"
    >
      <i
        aria-hidden="true"
        className="camera retro icon"
      />
      <span
        className="sc-bxivhb iVRQth"
      >
        2020 Nextlens.io
      </span>
    </a>
  </div>
  <div
    className="sc-bwzfXH cCmMUB"
  >
    <a
      className="sc-htpNat gEKCED"
      href="https://github.com/devonyu/nextLens-io"
    >
      <i
        aria-hidden="true"
        className="github icon"
      />
    </a>
    <a
      className="sc-htpNat gEKCED"
      href="https://www.facebook.com/nextlens.io/"
    >
      <i
        aria-hidden="true"
        className="facebook icon"
      />
    </a>
    <a
      className="sc-htpNat gEKCED"
      href="https://www.instagram.com/nextlens.io/"
    >
      <i
        aria-hidden="true"
        className="instagram icon"
      />
    </a>
    <a
      className="sc-htpNat gEKCED"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <i
        aria-hidden="true"
        className="code icon"
      />
    </a>
  </div>
</div>
`;
